src/SearchQueryService/Controllers/SearchController.cs

```diff
@@ -126,7 +128,52 @@ private static object CreateAzSearchResponse(AzPost newDocs)
         }
 
         private async Task PostDocuments(string indexName, AzPost docs)
-            => await _solrService.PostDocumentAsync(ConvertAzDocs(docs), indexName);
+        {
+            var delete = new AzPost { Value = new List<Dictionary<string, JsonElement>>() };
+            var addOrUpdate = new AzPost { Value = new List<Dictionary<string, JsonElement>>() };
+
+            foreach (var doc in docs.Value)
+            {
+                if (doc[_searchAction].GetString() == "delete")
+                {
+                    delete.Value.Add(doc);
+                }
+                else
+                {
+                    addOrUpdate.Value.Add(doc);
+                }
+            }
+
+            if (delete.Value.Count > 0)
+            {
+                await _solrService.DeleteDocumentsAsync(ConvertAzDocsForDelete(delete), indexName);
```

Comment:
What if this call fails? The next method will not be called. It is good? Wouldn't it be better to wrap both calls with a `try {} catch {}` and if it fails  throwing aggregate exception? I'm not sure myself.